File: src/clase-06/typescript/command/test_command.ts
import { Light, RemoteControl, LightOnCommand, LightOffCommand } from './command';

const light = new Light();
const rc = new RemoteControl();
rc.run(new LightOnCommand(light));
rc.run(new LightOffCommand(light));

if (rc.getHistory().length !== 2) throw new Error('Historial incorrecto');
if (light.state !== 'off') throw new Error('Estado final incorrecto');
console.log('test_command OK');

File: src/clase-06/python/command/test_command.py
from python.command.command import Light, RemoteControl, LightOnCommand, LightOffCommand

def test_command_history_and_effect(capsys):
    light = Light()
    rc = RemoteControl()

    rc.run(LightOnCommand(light))
    rc.run(LightOffCommand(light))

    assert len(rc.history) == 2
    assert light.state == "off"

    out = capsys.readouterr().out
    assert "Luz encendida" in out
    assert "Luz apagada" in out
